internal/pkg/manifest/validate_test.go

```diff
@@ -1491,7 +1516,135 @@ func TestRoutingRule_validate(t *testing.T) {
 	}
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
-			gotErr := tc.RoutingRule.validate()
+			gotErr := tc.ALBRoutingRule.validate()
+
+			if tc.wantedError != nil {
+				require.EqualError(t, gotErr, tc.wantedError.Error())
+				return
+			}
+			if tc.wantedErrorMsgPrefix != "" {
+				require.Error(t, gotErr)
+				require.Contains(t, gotErr.Error(), tc.wantedErrorMsgPrefix)
+				return
+			}
+			require.NoError(t, gotErr)
+		})
+	}
+}
+
+func TestRoutingRuleConfiguration_validate(t *testing.T) {
+	testCases := map[string]struct {
+		RoutingRuleConfiguration RoutingRuleConfiguration
+		wantedErrorMsgPrefix     string
+		wantedError              error
+	}{
+		"return if routing rule configuration is nil": {},
+		"error if both target_container and targetContainer are specified": {
+			RoutingRuleConfiguration: RoutingRuleConfiguration{
+				MainRoutingRule: ALBRoutingRule{
+					Path:            stringP("/"),
+					TargetContainer: aws.String("mockContainer"),
+				},
+				TargetContainerCamelCase: aws.String("mockContainer"),
+			},
+			wantedError: fmt.Errorf(`must specify one, not both, of "target_container" and "targetContainer"`),
+		},
+		"error if one of allowed_source_ips is not valid for additional routing rules": {
```

Comment:
Waht do you think of this instead:
1. Write unit test cases inside of `func TestALBRoutingRule_validate(t *testing.T)` to test if fields like `allowed_source_ips` and `protocol` are errored out
2.  Here, in addition to the first two test cases (confi is nil, targetContainer), just write two more unit test cases here: "error if the main routing rule is invalid", "error it an additional routing rule is invalid".